src/Compilers/CSharp/Test/Emit2/Semantics/ExtensionTypeTests.cs

```diff
@@ -16468,7 +16468,23 @@ public void M(int i)
 """;
         var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
 
-        CompileAndVerify(comp, expectedOutput: "E.M", verify: Verification.FailsPEVerify);
+        var verifier = CompileAndVerify(comp, expectedOutput: "E.M",
+            verify: Verification.FailsILVerify with { ILVerifyMessage = "[<Main>$]: Unrecognized arguments for delegate .ctor. { Offset = 0xb }" });
+
+        verifier.VerifyDiagnostics();
+
+        verifier.VerifyIL("<top-level-statements-entry-point>", """
+{
+  // Code size       24 (0x18)
+  .maxstack  2
+  IL_0000:  newobj     "C..ctor()"
+  IL_0005:  ldftn      "void E.M(int)"
+  IL_000b:  newobj     "D..ctor(object, nint)"
```

Comment:
This IL does look suspicious. We load an instance of `C` as a receiver, and method from `E`, which has no relationship with `C` from runtime point of view. Therefore, the IL doesn't look right. An addition of a prototype comment would be enough to unblock the merge.

Reply:
Yes, this is expected for all instance scenarios at the moment. Added PROTOTYPE